### ISSUE TITLE
[Bug] Can't download from apkmirror

### ISSUE THREAD

**Issue by gijsk** [mozilla-mobile/fenix#15914]
(Moved from https://bugzilla.mozilla.org/show_bug.cgi?id=1669119 )

## Steps to reproduce

Try downloading from APKMirror. I tried this one:
https://www.apkmirror.com/apk/google-inc/android-auto/android-auto-5-6-6034-release/android-auto-google-maps-media-messaging-5-6-603434-release-android-apk-download/download/

### Expected behavior

The file should download

### Actual behavior

You get prompted to download it. Tapping "Download" then apparently starts a download (the android notification tray shows a download item) but never completes - the download item in the system notification tray keeps an indeterminate state progress bar forever

### Device information

* Android device: unsure what device this was originally reported with, but this doesn't seem related.
* Fenix version: in https://bugzilla.mozilla.org/show_bug.cgi?id=1669119 the reporter tested with 81; I tried Firefox 82.0.0-beta-4 .

The report further notes that the download works in Chrome for Android - but only after a warning that the file "might harm the device". The download also works fine in Firefox Desktop.

**Comment by andreiuipath** [comment]
Why do you want to download it from APK mirror? You have Google Play, FF Updater from F-Droid https://f-droid.org/en/packages/de.marmaro.krt.ffupdater/ and the manual links https://firefox-ci-tc.services.mozilla.com/tasks/index/project.mobile.fenix.v2

**Comment by andreicristianpetcu** [comment]
Why do you want to download it from APK mirror? You have Google Play, FF Updater from F-Droid https://f-droid.org/en/packages/de.marmaro.krt.ffupdater/ and the manual links https://firefox-ci-tc.services.mozilla.com/tasks/index/project.mobile.fenix.v2

How is APK mirror better than these 3 options above?

The devs are overloaded with bugs and if there is a workaround for this issue maybe

**Comment by gijsk** [comment]
I'm just forwarding the issue from bugzilla, so I have no idea - though I suspect this happens with all APKs from apkmirror, not just Firefox. Regardless, the issue isn't so much the specific website, the issue is that the downloading UI breaks. It seems likely that this would happen elsewhere, too, and that it's related to headers, triggering principals, mixed content, safebrowsing or... something, that makes this category of downloads fail. A reproducible testcase, wherever it comes from, is then useful to narrow down the issue.

**Comment by Amejia481** [comment]
@gijsk thanks for the report, we did some improvement on nightly and I wasn't able to simulate the issue.If it's not much trouble would you mind trying it on [nightly](https://play.google.com/store/apps/details?id=org.mozilla.fenix)? I would ask to the QA team to double check too, as the have more devices for testing. Thanks in advance to both!

**Comment by gijsk** [comment]
@Amejia481 yep! It works for me on nightly. Thanks for all your work! (Unsure if you want me to close this out or wait for QA so I'll leave it as-is for now)

**Comment by Amejia481** [comment]
No problem, lets wait for QA to confirm :)  Credit ❤️  to @owlishDeveloper she did it all the hard work on the geckoView side!

**Comment by naiyerasif** [comment]
This is not a problem only with APKs. All the downloads are failing on Android 11. The same downloads work fine on Chrome or Chromium-based browsers. 

![image](https://user-images.githubusercontent.com/19614213/96291874-5c573780-1006-11eb-97a7-6aa118057275.png)

**Comment by Amejia481** [comment]
I was able to download in [nightly](https://play.google.com/store/apps/details?id=org.mozilla.fenix), could you try on nightly to confirm if it works for you?  :)

![image](https://user-images.githubusercontent.com/773158/96301816-bdaae680-0fc5-11eb-9ff1-b9437db74f21.png)

**Comment by naiyerasif** [comment]
Tried to download a couple of different files from different sites. All of them are failing. Could this be something specific to Android 11? I'm facing this on a OnePlus 8 Pro running Android 11.

**Comment by Amejia481** [comment]
Thanks, Is it failing on [nightly](https://play.google.com/store/apps/details?id=org.mozilla.fenix) ? Which version of nightly are you using? You can see by going to the three dotted menu -> Settings -> About Firefox Nighlty. It would be very helpful if you could share this print screen, thanks in advance!

**Comment by naiyerasif** [comment]
This issue is happening across all versions including stable (81.1.4), beta (82.0.0-beta.6) and nightly (201016 17:01).

**Comment by Amejia481** [comment]
Very odd, I'm using Android 11 too, with 201016 17:01, but I'm able to download. Could you confirm if your devices is missing the download folder? There is an open issue, that looks similar https://github.com/mozilla-mobile/fenix/issues/5829

**Comment by naiyerasif** [comment]
This indeed is happening because of the missing Download folder! Manually created the folder and tried the previously failing downloads. All of them got completed successfully. Shouldn't be Fenix creating the Download folder when it is missing, instead of outright failure?

**Comment by Amejia481** [comment]
> This indeed is happening because of the missing Download folder! Manually created the folder and tried the previously failing downloads. All of them got completed successfully. Shouldn't be Fenix creating the Download folder when it is missing, instead of outright failure?

Yes, we should, we are going to start working on this issue for [this sprint](https://github.com/orgs/mozilla-mobile/projects/3), sorry for all the inconveniences.

**Comment by LaurentiuApahideanSV** [comment]
Verified as fixed on Firefox Preview Nightly 201019 (Build #2015770539).

### Devices used:
- OnePlus 6T (Android 9)
- Samsung Galaxy S9 (Android 8.0.0)

**Comment by gabrielluong** [comment]
@LaurentiuApahideanSV not quite sure if this should be closed. @Amejia481 is this still something to look into?

**Comment by Amejia481** [comment]
I think is OK to close as the main issue is already fixed, and we another ticket for the pending work https://github.com/mozilla-mobile/fenix/issues/5829
